README.md
# Greasby Opticians 😎

<p align="center">
  <img width="500" src="src/assets/images/GO-logo.jpg" alt="Greasby Opticians Logo">
</p>

## Information about The Website

This website is the face of Greasby Opticians, a buisness in Greasby, England. I created the site using Vue js, Gridsome and Tailwind CSS.
See the [changelog](CHANGELOG.md) to see all updates to this project.

## Technologies Used

- **[Tailwind CSS](https://tailwindcss.com)**
  is a low-level CSS framework that allows you to build responsive css with html classes

- **[Gridsome](https://gridsome.org)**
  is a jamstack static site generator for vue, it allows you to use any cms, markdown and many forms of data to build websites that can be deployed on Github and hosted on [Netlify](https://www.netlify.com) or a similar host.

- **[Vue](https://vuejs.org)**
  is a JavaScript framework that is easy to use and build websites out of components and intergrate functionality into a site with ease.

## File Structure

```
.
|- package.json
|- gridsome.config.js
|- gridsome.server.js
|- prettier.config.js
|- tailwind.config.js
|- CHANGELOG.md
|- static/
  |- admin/
    |- config.yml
    |- index.html
  |- announcements/
    |- first-update.md, etc..
|- src/
  |- main.js
  |- main.css
  |- pages/
    |- Index.vue
    |- About.vue, etc..
  |- layouts/
    |- Default.vue
    |- any other layouts..
  |- components/
    |- default/
      |- Navigation.vue
      |- Footer.vue, etc..
    |- modules/
      |- News.vue
      |- OpenTimes.vue, etc..
  |- assets/
    |- images/
      |- logos, images, etc..
    |- styles/
      |- resets, global styles, etc..
|- dist/
  |- index.html
  |- 404.html
  |- css/ style pages...
  The dist folder is built when 'gridsome build' is ran
```

## Using Gridsome

### Install Gridsome CLI tool

- Using YARN: `yarn global add @gridsome/cli`
- Using NPM: `npm install --global @gridsome/cli`

### Create a Gridsome project

- `gridsome create my-gridsome-site` to create a new project
- `cd my-gridsome-site` to open folder
- `gridsome develop` to start local dev server at http://localhost:8080
- `gridsome build` to build a dist folder with generated static files

Your super fast and secure static site is ready to be deployed 🙌

---

[changelog](CHANGELOG.md)

<p>Created with 
<a href="https://gridsome.org" target="_blank" rel="noopener"><img height="16" src="src/assets/images/amykble/gridsome.svg" alt="gridsome"></a>
<a href="https://vuejs.org" target="_blank" rel="noopener"><img height="16" src="src/assets/images/amykble/vue.svg.png" alt="vue"></a>
<a href="https://tailwindcss.com" target="_blank" rel="noopener"><img height="16" src="src/assets/images/amykble/tailwind.png" alt="tailwind"></a>
 by <a href="#" target="_blank" rel="noopener">Amy Keable</a></p>


CHANGELOG.md
# Changelog 📝

All notable changes to this project will be documented in this file.

## [1.0.2] | 2020-07-15

#### Added

- Added About page photos

#### Changed

- x

#### Fixed

- Typo

---

## [1.0.1] | 2020-07-07

#### Added

- Added About page info

#### Changed

- x

#### Fixed

- x

---

## [1.0.0] | 2020-07-04

#### Added

- Added New Technology page

#### Changed

- x

#### Fixed

- x

---

## [0.16.1] | 2020-07-02

#### Added

- x

#### Changed

- Reworded VSC announcement

#### Fixed

- x

---

## [0.16.0] | 2020-07-01

#### Added

- Added the Contact Lenses page

#### Changed

- x

#### Fixed

- x

---

## [0.15.0] | 2020-07-01

#### Added

- Added the FAQ page

#### Changed

- x

#### Fixed

- x

---

## [0.14.0] | 2020-07-01

#### Added

- Added VSC and pdf

#### Changed

- x

#### Fixed

- x

---

## [0.13.9] | 2020-07-01

#### Added

- Added home page texts

#### Changed

- x

#### Fixed

- x

---

## [0.13.8] | 2020-06-30

#### Added

- Added new facts section

#### Changed

- x

#### Fixed

- x

---

## [0.13.7] | 2020-06-30

#### Added

- Added visual stress clinic link to footer and nav

#### Changed

- x

#### Fixed

- x

---

## [0.13.6] | 2020-06-29

#### Added

- Added visual stress clinic link

#### Changed

- x

#### Fixed

- x

---

## [0.13.5] | 2020-06-29

#### Added

- Added markdown image functionality

#### Changed

- x

#### Fixed

- x

---

## [0.13.4] | 2020-06-28

#### Added

- Added meta tags

#### Changed

- x

#### Fixed

- x

---

## [0.13.4] | 2020-06-28

#### Added

- x

#### Changed

- Created a Any Questions component

#### Fixed

- x

---

## [0.13.4] | 2020-06-27

#### Added

- x

#### Changed

- Changed images to links

#### Fixed

- x

---

## [0.13.3] | 2020-06-27

#### Added

- Added mail and tel links

#### Changed

- x

#### Fixed

- x

---

## [0.13.2] | 2020-06-25

#### Added

- x

#### Changed

- Adding announcement details

#### Fixed

- x

---

## [0.13.1] | 2020-06-24

#### Added

- x

#### Changed

- Arranging the announcements

#### Fixed

- x

---

## [0.13.0] | 2020-06-24

#### Added

- x

#### Changed

- Building the announcments pages

#### Fixed

- x

---

## [0.12.0] | 2020-06-23

#### Added

- Built the contact page

#### Changed

- x

#### Fixed

- x

---

## [0.11.0] | 2020-06-22

#### Added

- Built the frames page

#### Changed

- x

#### Fixed

- x

---

## [0.10.0] | 2020-06-21

#### Added

- Built the kids page

#### Changed

- x

#### Fixed

- x

---

## [0.9.1] | 2020-06-19

#### Added

- Added a favicon

#### Changed

- x

#### Fixed

- x

---

## [0.9.0] | 2020-06-18

#### Added

- Built the NHS page

#### Changed

- x

#### Fixed

- x

---

## [0.8.0] | 2020-06-18

#### Added

- Built the services page

#### Changed

- x

#### Fixed

- x

---

## [0.7.0] | 2020-06-17

#### Added

- Built the about page

#### Changed

- x

#### Fixed

- x

---

## [0.6.3] | 2020-06-16

#### Added

- x

#### Changed

- Seperation of concerns

#### Fixed

- x

---

## [0.6.2] | 2020-06-16

#### Added

- Added navigation large area for close

#### Changed

- Chnaged navigation animation

#### Fixed

- x

---

## [0.6.1] | 2020-06-16

#### Added

- x

#### Changed

- Created timed interval for reviews
- Changed colour of hamburger
- Made the review section background darker

#### Fixed

- x

---

## [0.6.0] | 2020-06-15

#### Added

- Added a review section
- Added a main body section to the homepage

#### Changed

- x

#### Fixed

- x

---

## [0.5.4] | 2020-06-13

#### Added

- x

#### Changed

- Chnaged updated slug to announcements

#### Fixed

- x

---

## [0.5.3] | 2020-06-11

#### Added

- x

#### Changed

- Added defer attribute to the Netlify index

#### Fixed

- x

---

## [0.5.2] | 2020-06-11

#### Added

- x

#### Changed

- Changed standard images for g-images

#### Fixed

- x

---

## [0.5.1] | 2020-06-11

#### Added

- Added responsive design to the footer

#### Changed

- x

#### Fixed

- Removed extra console logs

---

## [0.5.0] | 2020-06-10

#### Added

- Added a simple footer section

#### Changed

- x

#### Fixed

- Added 'gridsome build' to Netlify build command setting

---

## [0.4.0] | 2020-06-08

#### Added

- Added Vue Remark plugin
- Used the plugin to be the source of GraphQL data collection
- Set the collected data to display on the page

#### Changed

- x

#### Fixed

- x

---

## [0.3.2] | 2020-06-07

#### Added

- Creating Netlify cms connection

#### Changed

- x

#### Fixed

- x

---

## [0.3.1] | 2020-06-07

#### Added

- Added an announcement section on the homepage

#### Changed

- x

#### Fixed

- Fixed an issue with 'window is undefined' in the Navigation component

---

## [0.3.0] | 2020-06-05

#### Added

- Header area of the homepage, incuding the logo, background and button

#### Changed

- x

#### Fixed

- x

---

## [0.2.2] | 2020-06-05

#### Added

- x

#### Changed

- Set the navigation to be open on desktop screens and closed on phone screens on load

#### Fixed

- x

---

## [0.2.1] | 2020-06-03

#### Added

- x

#### Changed

- x

#### Fixed

- Removed src="main.css" from Navigation.vue, it was causing issues with the scoped css within the file

---

## [0.2.0] | 2020-06-03

#### Added

- The responsive navigation component
- Depoyed to a temporary doamin with Netlify

#### Changed

- Changed the font body to 'Quicksand' in the tailwind config file

#### Fixed

- x

---

## [0.1.0] | 2020-06-02

#### Added

- Initial project push to Github
- Added this changelog

#### Changed

- Formatted the README

#### Fixed

- x

---

This changlog format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/), and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).


tailwind.config.js
module.exports = {
  purge: ['./src/**/*.html', './src/**/*.vue', './src/**/*.jsx'],
  theme: {
    extend: {
      colors: {
        greasby: '#4B698D',
        'greasby-light': '#6082ab',
      },
      spacing: {
        '05': '0.125rem',
        '14': '3.5rem',
        '60': '15rem',
        '66': '16.5rem',
        '68': '17rem',
        '72': '18rem',
        '78': '19.5rem',
        '84': '21rem',
        '96': '24rem',
        '120': '30rem',
        '144': '36rem',
        '192': '48rem',
        'half-view': '50vh',
      },
      fontFamily: {
        body: ['Quicksand', 'Arial', 'sans-serif'],
        title: ['Tahoma', 'Arial', 'sans-serif'],
      },
      boxShadow: {
        r: '1px 0 3px 0 rgba(0, 0, 0, 0.1), 1px 0 2px 0 rgba(0, 0, 0, 0.06)', // right
        'r-md':
          '4px 0 6px -1px rgba(0, 0, 0, 0.1), 2px 0 4px -1px rgba(0, 0, 0, 0.06)', // right medium
        'r-lg':
          '10px 0 15px -3px rgba(0, 0, 0, 0.1), 4px 0 6px -2px rgba(0, 0, 0, 0.05)', // right large
      },
      borderRadius: {
        xl: '0.75rem',
        '2xl': '1rem',
        '3xl': '1.25rem',
        max: '100%',
      },
      opacity: {
        '10': '.1',
        '20': '.2',
        '30': '.3',
        '40': '.4',
        '60': '.6',
        '70': '.7',
        '80': '.8',
        '90': '.9',
      },
    },
  },
  variants: {},
  plugins: [],
}


gridsome.config.js
const tailwind = require('tailwindcss')
const postcssPlugins = [tailwind()]

module.exports = {
  siteName: 'Greasby Opticians',
  titleTemplate: '%s | Greasby Opticians',
  siteDescription:
    'Greasby Opticians is an independent opticians which offers excellent quality eyecare and an extensive range of spectacle frames and contact lenses. We have served the community of Greasby for over 30 years.',
  icon: {
    favicon: {
      src: './src/assets/images/GO-favicon.png',
      sizes: [16, 32, 96],
    },
  },
  plugins: [
    {
      use: '@gridsome/vue-remark',
      options: {
        typeName: 'Documentation',
        baseDir: './static/announcements',
        pathPrefix: 'announcements',
        template: './src/templates/AnnouncementsTemp.vue',
      },
    },
  ],
  css: {
    loaderOptions: {
      postcss: {
        plugins: postcssPlugins,
      },
    },
  },
}
